src/Search/Search.Management.Tests/Tests/SearchServiceTests.cs

```diff
@@ -184,6 +187,20 @@ public void CanScaleServiceUpAndDown()
             });
         }
 
+        [Fact]
+        public void CreateServiceWithInvalidNameGivesUsefulMessage()
+        {
+            Run(() =>
+            {
+                SearchManagementClient searchMgmt = GetSearchManagementClient();
+                SearchService service = DefineServiceWithSku(SkuName.Free);
+
+                CloudException e =
+                    Assert.Throws<CloudException>(() => searchMgmt.Services.CreateOrUpdate(Data.ResourceGroupName, InvalidServiceName, service));
+                Assert.Equal("Service name '----badname' is invalid: Service name must only contain lowercase letters, digits or dashes, cannot start or end with or contain consecutive dashes and is limited to 60 characters.", e.Message);
```

Comment:
You still have duplication of the InvalidServiceName, you could reference it in the assert here